rotkehlchen/api/rest.py

```diff
@@ -3906,3 +3907,48 @@ def get_associated_locations(self) -> Response:
             result=_wrap_in_ok_result([str(location) for location in locations]),
             status_code=HTTPStatus.OK,
         )
+
+    def _query_kraken_staking_events(
+        self,
+        only_cache: bool,
+        filter_query: HistoryEventFilterQuery,
+    ) -> Dict[str, Any]:
+        exchanges_list = self.rotkehlchen.exchange_manager.connected_exchanges.get(
+            Location.KRAKEN,
+        )
+        if exchanges_list is None:
+            events = []
+        else:
+            if only_cache is False:
+                for kraken_instance in exchanges_list:
+                    kraken_instance.query_kraken_ledgers(   # type: ignore
```

Comment:
Can't this raise? If it can it needs to be checked.

Also from what I see perhaps we catch all in there and return False in case of error. That `False` needs to be checked for an an appropriate API error returned.

Reply:
I've added a message so now the frontend can know if the API failed and returned only database entries. Also during the testing with real data saw that a catch in the kraken class was missing for remoterror. This function shouldn't raise any error.